Issue found by the random tester 2

```python
___________________________________________________________ Test_Undo.test_undo_random_insertdeleteundo_slow ___________________________________________________________

self = <test.test_eco.Test_Undo instance at 0x7fffc6247b90>

    @slow
    def test_undo_random_insertdeleteundo_slow(self):
>       self.random_insert_delete_undo(programs.connect4)

test/test_eco.py:2175:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/test_eco.py:2285: in random_insert_delete_undo
    x = self.treemanager.key_delete()
treemanager.py:1011: in key_delete
    self.delete_linebreak(self.cursor.line, node)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <treemanager.TreeManager object at 0x7fffc129bd50>, y = 0, node = TextNode(Terminal(''\r''), 63, 0, <return>)

    def delete_linebreak(self, y, node):
        deleted = self.lines[y+1].node
>       assert deleted is node
E       assert TextNode(Terminal(''\r''), 63, 0, <return>) is TextNode(Terminal(''\r''), 63, 0, <return>)

treemanager.py:1462: AssertionError
------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------
self.treemanager.cursor_reset()
self.move(DOWN, 56)
self.move(RIGHT, 0)
self.treemanager.key_delete()
self.treemanager.cursor_reset()
self.move(DOWN, 56)
self.move(RIGHT, 1)
self.treemanager.key_normal('u')
self.treemanager.cursor_reset()
self.move(DOWN, 56)
self.move(RIGHT, 4)
self.treemanager.key_normal('i')
self.treemanager.cursor_reset()
self.move(DOWN, 56)
self.move(RIGHT, 3)
self.treemanager.key_ctrl_z()
self.treemanager.cursor_reset()
self.move(DOWN, 56)
self.move(RIGHT, 2)
self.treemanager.key_ctrl_z()
self.treemanager.cursor_reset()
self.move(DOWN, 46)
self.move(RIGHT, 0)
self.treemanager.key_normal('\r')
self.treemanager.cursor_reset()
self.move(DOWN, 46)
self.move(RIGHT, 1)
self.treemanager.key_normal('!')
self.treemanager.cursor_reset()
self.move(DOWN, 46)
self.move(RIGHT, 3)
self.treemanager.key_normal('n')
self.treemanager.cursor_reset()
self.move(DOWN, 46)
self.move(RIGHT, 2)
self.treemanager.key_ctrl_z() 
self.treemanager.cursor_reset()
self.move(DOWN, 46)
self.move(RIGHT, 4)
self.treemanager.key_normal('a')
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 11)
self.move(RIGHT, 4)
self.treemanager.key_delete() 
self.treemanager.cursor_reset()
self.move(DOWN, 11)
self.move(RIGHT, 3)
self.treemanager.key_normal('9')
self.treemanager.cursor_reset()
self.move(DOWN, 11)
self.move(RIGHT, 1)
self.treemanager.key_normal('(')
self.treemanager.cursor_reset()
self.move(DOWN, 11)
self.move(RIGHT, 2)
self.treemanager.key_normal('^')
self.treemanager.cursor_reset()
self.move(DOWN, 11)
self.move(RIGHT, 0)
self.treemanager.key_normal('.')
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 50)
self.move(RIGHT, 1)
self.treemanager.key_normal('h')
self.treemanager.cursor_reset()
self.move(DOWN, 50)
self.move(RIGHT, 4)
self.treemanager.key_normal('}')
self.treemanager.cursor_reset()
self.move(DOWN, 50)
self.move(RIGHT, 0)
self.treemanager.key_normal('g')
self.treemanager.cursor_reset()
self.move(DOWN, 50)
self.move(RIGHT, 2)
self.treemanager.key_normal('n')
self.treemanager.cursor_reset()
self.move(DOWN, 50)
self.move(RIGHT, 3)
self.treemanager.key_ctrl_z() 
self.treemanager.cursor_reset()
self.move(DOWN, 60)
self.move(RIGHT, 4)
self.treemanager.key_normal(' ')
self.treemanager.cursor_reset()
self.move(DOWN, 60)
self.move(RIGHT, 2)
self.treemanager.key_normal('6')
self.treemanager.cursor_reset()
self.move(DOWN, 60)
self.move(RIGHT, 1)
self.treemanager.key_delete() 
self.treemanager.cursor_reset()
self.move(DOWN, 60)
self.move(RIGHT, 0)
self.treemanager.key_normal('o')
self.treemanager.cursor_reset()
self.move(DOWN, 60)
self.move(RIGHT, 3)
self.treemanager.key_delete() 
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 44)
self.move(RIGHT, 2)
self.treemanager.key_normal('l')
self.treemanager.cursor_reset()
self.move(DOWN, 44)
self.move(RIGHT, 4)
self.treemanager.key_normal('y')
self.treemanager.cursor_reset()
self.move(DOWN, 44)
self.move(RIGHT, 0)
self.treemanager.key_normal('[')
self.treemanager.cursor_reset()
self.move(DOWN, 44)
self.move(RIGHT, 3)
self.treemanager.key_delete() 
self.treemanager.cursor_reset()
self.move(DOWN, 44)
self.move(RIGHT, 1)
self.treemanager.key_normal('5')
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 17)
self.move(RIGHT, 4)
self.treemanager.key_normal('e')
self.treemanager.cursor_reset()
self.move(DOWN, 17)
self.move(RIGHT, 1)
self.treemanager.key_normal('v')
self.treemanager.cursor_reset()
self.move(DOWN, 17)
self.move(RIGHT, 2)
self.treemanager.key_normal('5')
self.treemanager.cursor_reset()
self.move(DOWN, 17)
self.move(RIGHT, 0)
self.treemanager.key_normal('d')
self.treemanager.cursor_reset()
self.move(DOWN, 17)
self.move(RIGHT, 3)
self.treemanager.key_normal('g')
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 21)
self.move(RIGHT, 4)
self.treemanager.key_ctrl_z() 
self.treemanager.cursor_reset()
self.move(DOWN, 21)
self.move(RIGHT, 1)
self.treemanager.key_normal('7')
self.treemanager.cursor_reset()
self.move(DOWN, 21)
self.move(RIGHT, 0)
self.treemanager.key_normal(')')
self.treemanager.cursor_reset()
self.move(DOWN, 21)
self.move(RIGHT, 3)
self.treemanager.key_normal('&')
self.treemanager.cursor_reset()
self.move(DOWN, 21)
self.move(RIGHT, 2)
self.treemanager.key_normal('j')
self.treemanager.undo_snapshot()
self.treemanager.cursor_reset()
self.move(DOWN, 0)
self.move(RIGHT, 3)
self.treemanager.key_normal('y')
self.treemanager.cursor_reset()
self.move(DOWN, 0)
self.move(RIGHT, 1)
self.treemanager.key_normal('\r')
self.treemanager.cursor_reset()
self.move(DOWN, 0)
self.move(RIGHT, 0)
self.treemanager.key_normal('\r')
self.treemanager.cursor_reset()
self.move(DOWN, 0)
self.move(RIGHT, 4)
self.treemanager.key_normal('4')
self.treemanager.cursor_reset()
self.move(DOWN, 0)
self.move(RIGHT, 2)
self.treemanager.key_delete() 
=========================================================== 1 failed, 84 passed, 8 skipped in 91.86 seconds ============================================================
```

--- a/lib/eco/test/test_eco.py
+++ b/lib/eco/test/test_eco.py
@@ -2251,6 +2251,29 @@ class Test_Undo(Test_Python):
         self.treemanager.key_ctrl_z()
         self.text_compare(prog)
 
+    def test_undo_random_insertdeleteundo_bug3(self):
+        self.reset()
+
+        self.treemanager.import_file("""class C:
+    x = 5
+""")
+        assert self.parser.last_status == True
+
+        self.treemanager.cursor_reset()
+        self.move(DOWN, 0)
+        self.move(RIGHT, 1)
+        self.treemanager.key_normal('\r')
+        self.treemanager.cursor_reset()
+        self.move(DOWN, 0)
+        self.move(RIGHT, 0)
+        self.treemanager.key_normal('\r')
+        self.treemanager.cursor_reset()
+        self.move(DOWN, 0)
+        self.move(RIGHT, 2)
+        self.treemanager.key_delete()
+
+        assert self.parser.last_status == False
+
     def random_insert_delete_undo(self, program):
         import random
         self.reset()

--- a/lib/eco/treemanager.py
+++ b/lib/eco/treemanager.py
@@ -117,6 +117,8 @@ class Cursor(object):
         if not node is self.node:
             self.node = node
             self.pos = 0
+            if node.symbol.name == "\r":
+                self.line += 1
         if self.pos < len(self.node.symbol.name):
             self.pos += 1
         else:

